middleware/lib/FunctionInvocationRequest.ts

```diff
@@ -0,0 +1,87 @@
+
+import {Logger} from '@salesforce/core/lib/logger';
+import {SObject} from '@salesforce/salesforce-sdk/dist/objects';
+import {
+    DataApi,
+    ErrorResult,
+    SuccessResult,
+} from '@salesforce/salesforce-sdk/dist/api';
+
+export enum FunctionInvocationRequestStatusEnum {
+  Success = 'Success',
+  Error = 'Error'
+}
+
+// Save update to FunctionInvocationRequest
+// Does not throw
+export async function saveFnInvocation(logger: Logger, 
+                                       fnInvocation: FunctionInvocationRequest, 
+                                       response: any, 
```

Comment:
I'm a bit concerned about `response` here. This could be a string, object, or even a promise. Today we only handle if it is falsely, but now we are trying to log the response back to core. We may need to test to see if it is a promise and handle that properly. As of right now, I think the `save` code below would essentially serialize the promise to a string.